unstructured_inference/models/yolox_model.py

```diff
@@ -0,0 +1,130 @@
+import os
+import tempfile
+
+import cv2
+import jsons
+import numpy as np
+import onnxruntime
+from pdf2image import convert_from_path
+
+import unstructured_inference
+from unstructured_inference.inference.layout import DocumentLayout, LayoutElement, PageLayout
+from unstructured_inference.visualize import vis
+from unstructured_inference.yolox_functions import demo_postprocess, multiclass_nms
+from unstructured_inference.yolox_functions import preproc as preprocess
+from unstructured_inference.models import _get_model_loading_info
+
+
+def yolox_local_inference(
+    filename: str, type: str = "image", to_json: bool = False, keep_output: bool = False
```

Comment:
Instead of `keep_output: bool`, could we have `output_directory: Optional[str]`? Then the user can specify the directory, and if it's `None` that's the equivalent to what `keep_output=False` does now. We can also get rid of the `OUTPUT_DIR` constance in that case.

I'd also drop the `to_json` boolean and let the user do that themselves if they want to. That way you don't always have to do a type check on the output of this function.

Reply:
The result is so elegant! Thanks for the suggestions 😄